src/dmd/dmodule.d

```diff
@@ -1035,6 +1054,21 @@ extern (C++) final class Module : Package
         {
             isHdrFile = true;
         }
+
+        /* If it has the extension ".c", it is a "C" file.
+         */
+        if (FileName.equalsExt(arg, c_ext))
+        {
+            isCFile = true;
+
+            scope p = new CParser!AST(this, buf, cast(bool) docfile,
+                cast(ubyte) target.c.longsize, cast(ubyte) target.c.long_doublesize, cast(ubyte) target.c.twchar_t.size());
```

Comment:
Could you instead pass a `const` copy/reference to `target.c`, this would allow me to quickly add the other types immediately after this gets pulled.

Reply:
I wanted cparse.d to have minimal dependencies on anything else.

Reply:
What you could do is move TargetC to globals.d, just the data fields, not the member functions. That would sever its dependencies. Change twchar_t to a size rather than a Type, then the struct can stand alone.

Reply:
CParser is already a template though, so the cparse.d module need not import `dmd.target`.